src/TensorFlowNET.Keras/Engine/DataAdapters/TensorLikeDataAdapter.cs

```diff
@@ -20,7 +21,7 @@ public TensorLikeDataAdapter(DataAdapterArgs args)
         {
             this.args = args;
             _process_tensorlike();
-            num_samples = (int)args.X.shape[0];
+            num_samples = (int)args.X[0].shape[0];
```

Comment:
Can we use `(int)args.X.shape[0]` instead of `(int)args.X[0].shape[0]`?

Reply:
Sure, I have changed it.